### ISSUE TITLE
GraphQL - Ability to select a link that references the objects detail page

### ISSUE THREAD

**Issue by jifox** [nautobot/nautobot#1939]
### As ...

Nelly - Network Engineer

### I want ...

to be able to edit objects that are displayed as GraphQL result.

### So that ...

I can easily edit all the filtered objects without having to manually select the displayed records in Nautobot.

### I know this is done when...

- I execute a GraphQL Query
    ```
    {
      "data": {
        "interfaces": [
          {
            "device": {
              "name": "ATBRACSE010"
            },
            "name": "GigabitEthernet1/0/42",
            "description": "1/0/42 to Lichtsteuerung",
            "id": "4778126a-6467-4023-a813-e2c97a67c1e1",
            "obj": a link to: https://nautobot.local:8083/dcim/interfaces/4778126a-6467-4023-a813-e2c97a67c1e1
          },
    ````
- Use "obj" as field name to return the link to the "detail page"

### Optional - Feature groups this request pertains to.

- [ ] Automation
- [ ] Circuits
- [ ] DCIM
- [ ] IPAM
- [ ] Misc (including Data Sources)
- [ ] Organization
- [ ] Plugins (and other Extensibility)
- [ ] Security (Secrets, etc)
- [ ] Image Management
- [X] UI/UX
- [ ] Documentation
- [ ] Other (not directly a platform feature)

**Comment by glennmatthews** [comment]
Possible approach could be adding an automatic `obj` or `url` field to all graphql types that simply calls `object.get_absolute_url()`.

**Comment by jifox** [comment]
@glennmatthews 

My suggestion is:
obj:    object.get_absolute_url()   --> Display in GraphQL as Hyperlink (new Tab)
url:     object.get_absolute_url()   --> Just as in other API results

**Comment by glennmatthews** [comment]
Hm, that does raise a good point. Do we want the link to be a link for the UI (`/dcim/sites/my-site-slug/`) or a link for the REST API (`/api/dcim/sites/<UUID>/`)? `get_absolute_url()` generally returns the UI link.

**Comment by jifox** [comment]
Yes, I want it primarily for that. So I can use GraphQL to select all the things that I like to edit. Also in API it will allow edit records

**Comment by bryanculver** [comment]
Related: #4163
